### DISCUSSION TITLE
[Fixed] Moving klipper to the stub ccm broke our project

### DISCUSSION THREAD

**Discussion by mysticaltech** [k3s-io/k3s#6452] [General]
Hello @brandond,

Just FYI, moving klipper to the ccm (as explained [here](https://github.com/k3s-io/k3s/blob/027039530d43bc01c9c17cc0be48db583bff943f/docs/adrs/servicelb-ccm.md)) it broke one of our feature on [kube-hetzner](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/397), because we are already using the hetzner ccm, and enabling klipper now breaks the cluster deployment, as port 10258 is already used by the stub ccm for klipper.

What a shame really! Now we have to changes lots of things in order to use metallb instead.

**Comment by brandond** [comment]
You know you can change the hetzner cloud-controller's listener port with the `--secure-port` option, or disable it entirely by setting it to 0, right? All it does is serve a few prometheus metrics so it's hardly important.

```console
brandond@dev01:~$ docker run --rm -it hetznercloud/hcloud-cloud-controller-manager:v1.13.2 --help 2>&1 | grep secure-port
      --bind-address ip                        The IP address on which to listen for the --secure-port port. The associated interface(s) must be reachable by the rest of the cluster, and by CLI/web clients. If blank or an unspecified address (0.0.0.0 or ::), all interfaces will be used. (default 0.0.0.0)
      --secure-port int                        The port on which to serve HTTPS with authentication and authorization. If 0, don't serve HTTPS at all. (default 10258)
```

I'll open an issue to make the k3s cloud-controller manager args configurable so that you can disable or reconfigure our port instead, if you prefer. In the mean time simply changing the hetzner port seems like a reasonable workaround.

**Reply by mysticaltech** [reply]
@brandond I did not know; thank you for the info! 🙏 I will try modifying the --secure-port for now! 👍👍

**Reply by mysticaltech** [reply]
@brandond  Confirmed working!  I used `--secure-port=10288` (I chose that number as not used anywhere else in Kube). 

Thanks again for that quick tip! 🙏

![ksnip_20221106-014245](https://user-images.githubusercontent.com/518555/200148821-9234c840-f184-4ff2-be3e-bd6eeecbcf4f.png)

**Reply by brandond** [reply]
I forgot that we actually do already allow configuring the CCM args. You can start k3s with `--kube-cloud-controller-manager-arg=secure-port=0` to prevent K3s from occupying that port.

**Reply by mysticaltech** [reply]
Oh wow, simpler indeed! Thank you 🙏🙏
